pkg/utils/utils.go

```diff
@@ -110,6 +110,12 @@ func ErrorContains(out error, want string) bool {
 // We MUST NOT check the name or key property here. It MAY be defaulted by the provider.
 func ValidateSecretSelector(store esv1beta1.GenericStore, ref esmeta.SecretKeySelector) error {
 	clusterScope := store.GetObjectKind().GroupVersionKind().Kind == esv1beta1.ClusterSecretStoreKind
+	if ref.Name == "" {
```

Comment:
see above:
```
// We MUST NOT check the name or key property here.
// It MAY be defaulted by the provider.
```

AFAIK provider-level defaulting is only implemented by the vault provider

Reply:
We added this here because if we did not, we could get some panicking happening, but we just skipped through the comments :sweat:. I'll put this logic into IBM provider itself.